src/ophyd_async/testing/_assert.py

```diff
@@ -69,26 +73,39 @@ def _approx_reading(expected: Mapping[str, Any], actual: Reading) -> Reading:
 
 
 def _assert_readings_approx_equal(
-    expected: Mapping[str, Mapping[str, Any]], actual: Mapping[str, Reading]
+    expected: Mapping[str, Mapping[str, Any]],
+    actual: Mapping[str, Reading],
+    full_match: bool = True,
 ):
-    assert actual == {
-        k: _approx_reading(v, actual[k]) for k, v in expected.items() if k in actual
+    if full_match:
+        assert expected.keys() == actual.keys()
+    else:
+        assert expected.keys() <= actual.keys()
+    # expected keys are sub- or full set of actual keys
+    assert {k: actual[k] for k in expected.keys()} == {
+        k: _approx_reading(v, actual[k]) for k, v in expected.items()
```

Comment:
The diff might be a little confusing as it only shows the expected items rather than the actual ones. If we expand to include the actual values, only overwritten by expected if they exist, would that work?
```suggestion
    assert actual == {
        k: _approx_reading(expected.get(k, v), v) for k, v in actual.items()
```

Reply:
Yes this will work if we leave the new check for keys subset/fullset above in place.